File: NDPServer/src/main/java/com/slender/config/manager/ValidatorManager.java
package com.slender.config.manager;

import com.slender.exception.category.ValidationException;
import jakarta.validation.ConstraintViolation;
import jakarta.validation.Validator;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.util.Set;

@Component
@RequiredArgsConstructor
public final class ValidatorManager {
    private final Validator validator;
    public <T> void validate(T validationObject){
        Set<ConstraintViolation<T>> validateMessages = validator.validate(validationObject);
        if (!validateMessages.isEmpty()) {
            String message = validateMessages.stream()
                    .map(ConstraintViolation::getMessage)
                    .toList()
                    .getFirst();
            throw new ValidationException(message);
        }
    }
}

File: NDPServer/src/main/java/com/slender/config/manager/ValidatorManager.kt
package com.slender.config.manager

import com.slender.exception.category.ValidationException
import jakarta.validation.ConstraintViolation
import jakarta.validation.Validator
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Component

@Component
@RequiredArgsConstructor
class ValidatorManager(
    private val validator: Validator
) {
    fun <T> validate(validationObject: T) {
        val validateMessages = validator.validate(validationObject)
        if (!validateMessages.isEmpty()) {
            val message = validateMessages.stream()
                .map { obj: ConstraintViolation<T> -> obj.message }
                .toList()
                .first()
            throw ValidationException(message)
        }
    }
}
